### DISCUSSION TITLE
Docker action uses build instead of pull

### DISCUSSION THREAD

**Discussion by jdbruijn** [community/community#25175] [Actions]
<p>I’ve recently created a GitHub Action using docker: <a href="https://github.com/vidavidorra/github-action-renovate" rel="nofollow noopener">https://github.com/vidavidorra/github-action-renovate</a><br>
Where I use this action, I see in the logs that it is buiding the action, rather than pulling it like I see with other actions. For example see <a href="https://github.com/vidavidorra/github-renovate/runs/536319505?check_suite_focus=true" rel="nofollow noopener">this build log</a>, where the action is being buit. In the <a href="https://github.com/vidavidorra/github-action-renovate/runs/533875750?check_suite_focus=true" rel="nofollow noopener">log</a> of an 3rd party action I see that it is pulling the image.<br>
Why is my action not being pulled, but built every time? I see that the time penalty is about 1 minute with the pulled action vs buidling. I searched the docs, but couldn’t find information on it. Do I need to push the image to Docker Hub in order to be able to pull the image? If so, does GitHub Actions automatically detect that it can pull the image and doesn’t need to build it?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/jdbruijn">@jdbruijn</a> , </p>
<p>I checked <a href="https://github.com/wagoid/commitlint-github-action" rel="nofollow noopener">wagoid/commitlint-github-action</a>.  In its action.yml , the author specific a <a href="https://hub.docker.com/r/wagoid/commitlint-github-action" rel="nofollow noopener">docker image</a> in dockerhub under  <strong>runs </strong> key. When using this action in a workflow, it will pull the image from dockerhub. </p>
<pre><code>runs:
  using: 'docker'
  image: 'docker://wagoid/commitlint-github-action:1.6.0'
</code></pre>
<p>And in your action.yml , you are using Dockerfile. When using this action in a workflow, it will build the image using the Dockerfile. </p>
<pre><code>runs:
  using: docker
  image: src/Dockerfile
</code></pre>
<p> If you want to pull image instead of build when using your Docker action , I would recommend you push the image into Dockerhub and then modify your action.yml &gt; runs &gt; image using an image instead of Dockerfile.  </p>

**Comment by jdbruijn** [comment]
<p>I see, thanks for the explanation!</p>
<p>I’ll have a look then whether I want to setup Docker build and push in my workflow. Would make it a bit more complex since I’d have to build that before I can test the action, and then also would have to update my action.yml based on that build. And I want to do all that automatically in my workflow :slight_smile:<br>
The docker build and push wouldn’t be such an issue, but the updating of action.yml would probably be since I also run self-tests in my action.</p>
<p>Anyways,  I’ll have to look into all that further. Thanks for the answer!</p>
